rust-lang/cargo
# Stabilize `-Zmultitarget`

### What does this PR try to resolve?

Stabilize `-Zmultitarget`. Closes #8176.

`-Zmultitarget` exists on nightly for years (#8167, merged in April 2020). It has already been in the wild for a long time, and is the main code path of cargo compilation since then. It may be the time to push it to stable.

### How should we test and review this PR?

Almost all build commands get this update, only `cargo run` and `cargo install` accept no more than one `--target`. Here is a list of commands that support multiple `--target` options:

- `cargo bench`
- `cargo build`
- `cargo check`
- `cargo clean`
- `cargo doc`
- `cargo fetch`
- `cargo fix`
- `cargo package`
- `cargo publish`
- `cargo rustc`
- `cargo rustdoc`
- `cargo test`

You may want to check if anything weird for them to support multitarget. Also get a glimpse on their docs.

### Additional information

I found that `cargo clean --target` does not work. I'll look into it and post another issue afterward.
